internal/ingress/controller/template/template.go

```diff
@@ -412,8 +412,8 @@ func buildLoadBalancingConfig(b interface{}, fallbackLoadBalancing string) strin
 		return ""
 	}
 
-	if backend.UpstreamHashBy != "" {
-		return fmt.Sprintf("hash %s consistent;", backend.UpstreamHashBy)
+	if backend.UpstreamHashBy.UpstreamHashBy != "" {
+		return fmt.Sprintf("hash %s consistent;", backend.UpstreamHashBy.UpstreamHashBy)
```

Comment:
this function is dead, please rebase once https://github.com/kubernetes/ingress-nginx/pull/3478 gets merged